test/dev/ui/spinner_test.rb

```diff
@@ -60,19 +89,11 @@ def match_lines(out, *patterns)
         # newline, or cursor-down
         lines = out.split(/\n|\x1b\[\d*B/)
 
-        # Assert all lines are matched
-        lines.each do |l|
```

Comment:
This doesn't  catch extra output. I can output as many lines as I want as long as I match one of the patterns. I added `assert_equal patterns.size, lines.size` which does catch extra output